src/modules/commander/px4_custom_mode.h

```diff
@@ -66,6 +66,11 @@ enum PX4_CUSTOM_SUB_MODE_AUTO {
 	PX4_CUSTOM_SUB_MODE_AUTO_PRECLAND
 };
 
+enum PX4_CUSTOM_SUB_MODE_POSCTL {
+	PX4_CUSTOM_SUB_MODE_POSCTL_POSCTL = 1,
```

Comment:
I don't think this change is backward compatible. Previously there were no sub modes for POSCTL hence a POSCTL flight mode was indicated by main_mode = PX4_CUSTOM_MAIN_MODE_POSCTL and sub_mode = 0. This new header seems to change this such that now POSCTL is identified by sub_mode = 1.

Shouldn't PX4_CUSTOM_SUB_MODE_POSCTL_POSCTL = 0?

Reply:
@MaEtUgR FYI

Reply:
So I verified using this pull and QGC that the following is what is sent:
POSCTL - sub_mode = 0
ORBIT - sub_mode = 2

Which is good from a back compat standpoint but not what the headers says. I'm going to adjust the QGC header to this:
```
enum PX4_CUSTOM_SUB_MODE_POSCTL {
    PX4_CUSTOM_SUB_MODE_POSCTL_POSCTL = 0,
    PX4_CUSTOM_SUB_MODE_POSCTL_ORBIT
};
```

Under the assumption that is what the firmware will do as well.

Reply:
Sure, let's do that. Sorry I didn't pay enough attention to the numbers because I thought it's a shared header anyways but backwards compatibility is of course mandatory in this case. I'm glad if it works without adding a main mode (possible as last resort).

Reply:
@DonLakeFlyer I force-pushed `PX4_CUSTOM_SUB_MODE_POSCTL_POSCTL = 0`